python/src/iceberg/utils/openapi.py

```diff
@@ -0,0 +1,32 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import os
+from functools import lru_cache
+
+import yaml
+
+from iceberg import assets
+
+
+@lru_cache(maxsize=1)
```

Comment:
Once https://github.com/apache/iceberg/pull/4784 has been merged, we can simplify this:
```suggestion
@cache
```
https://docs.python.org/3/library/functools.html#functools.cache